snapcraft/internal/project_loader/_parts_config.py

```diff
@@ -187,7 +187,7 @@ def load_part(self, part_name, plugin_name, part_properties):
 
         part = pluginhandler.PluginHandler(
             plugin=plugin,
-            part_properties=part_properties,
+            part_properties=grammar_processor.get_properties(),
```

Comment:
This is still going to mess up state tracking. The properties here need to be the raw ones. Note how we pass the grammar processor to the PluginHandler anyway-- it should just use the processor to grab the source like it does for e.g. stage-packages.